src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private static final int INDEX_NULL = 0;
+    private Object [] array;
+    private int arrayLength;
+    private int elementsAmount;
+
+    public ArrayList() {
+        arrayLength = DEFAULT_ARRAY_SIZE;
+        array = new Object [arrayLength];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementsAmount == arrayLength) {
+            resize(arrayLength);
+        }
+        array[elementsAmount] = value;
+        elementsAmount++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > elementsAmount || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound");
+        }
+        Object[]newArray = new Object[arrayLength];
+        System.arraycopy(array, INDEX_NULL, newArray, INDEX_NULL, elementsAmount);
+        System.arraycopy(array, index, newArray, index + 1, elementsAmount - index);
+        newArray[index] = value;
+        array = newArray;
+        elementsAmount++;
+        if (elementsAmount == arrayLength) {
+            resize(arrayLength);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if ((list.size() + elementsAmount) > arrayLength) {
+            resize(elementsAmount + list.size());
+        }
+        for (int i1 = 0; i1 < list.size(); i1++) {
+            array[elementsAmount] = list.get(i1);
+            elementsAmount++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= elementsAmount || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound");
+        }
```

Comment:
move this check to private method checkIndex(int index)
```suggestion
        if (index >= elementsAmount || index < 0) {
            throw new ArrayListIndexOutOfBoundsException("Index is out of bound: " + index);
        }
```